Orange/widgets/gui.py

```diff
@@ -886,10 +887,13 @@ def lineEdit(widget, master, value, label=None, labelWidth=None,
     if validator:
         ledit.setValidator(validator)
     if value:
+        if valueType is None:
+            current_value = getdeepattr(master, value)
+            valueType = str if current_value is None else type(current_value)
```

Comment:
Revert to the old default if the `current_value` is `None`. Otherwise, the `valueType` must match the type of `current_value`; if any widget kept the default `valueType` (`str`) but `current_value` was something else (that is, `int` or `float`), the widget was already broken.